packages/program-boilerplate/src/conversion.ts

```diff
@@ -66,6 +69,47 @@ function parseValue(value: string): RuleValue {
   return value;
 }
 
+/**
+ * Checks whether any of the edge trigger conditions are met given
+ * a list of fields to check and the current context.
+ *
+ * @param {string[] | undefined} fields The list of edge trigger fields
+ * @param {any} activeTrigger The current program `activeTrigger`
+ *
+ * @return {boolean} Whether any of the edge fields have changed
+ */
+export function meetEdgeTriggerConditions(
+  fields: string[] | undefined,
+  activeTrigger: any
+): boolean {
+  if (fields === undefined || fields.length === 0) {
+    return true;
+  }
+
+  for (const field of fields) {
+    if (!field.startsWith("user.")) {
+      // TODO: what to do here? this is probably some kind of error case
```

Comment:
@jayden-chan I think the right thing to do here might be to fail fast and return `false`. There's obviously some kind of condition trying to be attempted, but we can't process it so denying the trigger feels like the right thing to do.

Please make sure there is a unit test for this case.